apps/wallet-dashboard/app/(protected)/assets/page.tsx

```diff
@@ -27,34 +25,46 @@ const ASSET_CATEGORIES: { label: string; value: AssetCategory }[] = [
 
 export default function AssetsDashboardPage(): React.JSX.Element {
     const [selectedAsset, setSelectedAsset] = useState<IotaObjectData | null>(null);
-    const [selectedCategory, setSelectedCategory] = useState<AssetCategory>(AssetCategory.Visual);
+    const [selectedCategory, setSelectedCategory] = useState<AssetCategory | null>(null);
     const account = useCurrentAccount();
-    const { data, isFetching, fetchNextPage, hasNextPage, refetch } = useGetOwnedObjects(
-        account?.address,
-        undefined,
-        OBJECTS_PER_REQ,
-    );
+    const {
+        data: ownedAssets,
+        isFetching,
+        fetchNextPage,
+        hasNextPage,
+        refetch,
+    } = useGetNFTs(account?.address);
 
-    const assets: IotaObjectData[] = [];
+    const assets: IotaObjectData[] = (() => {
+        if (selectedCategory === null) return [] as IotaObjectData[];
 
-    for (const page of data?.pages || []) {
-        for (const asset of page.data) {
-            if (asset.data && asset.data.objectId) {
-                if (selectedCategory == AssetCategory.Visual) {
-                    if (hasDisplayData(asset)) {
-                        assets.push(asset.data);
-                    }
-                } else if (selectedCategory == AssetCategory.Other) {
-                    assets.push(asset.data);
-                }
+        const assetsList = ownedAssets ? ownedAssets[selectedCategory] : [];
+        return assetsList.filter((asset) => {
+            if (selectedCategory === AssetCategory.Visual) {
+                return hasDisplayData({ data: asset });
             }
-        }
-    }
+            return true;
+        });
+    })();
 
     function onAssetClick(asset: IotaObjectData) {
         setSelectedAsset(asset);
     }
 
+    useEffect(() => {
+        if (!ownedAssets || selectedCategory !== null) {
+            return;
+        }
+
+        const defaultCategory =
+            ownedAssets.visual.length > 0
+                ? AssetCategory.Visual
+                : ownedAssets.other.length > 0
+                  ? AssetCategory.Other
+                  : AssetCategory.Visual;
+        setSelectedCategory(defaultCategory);
+    }, [ownedAssets, selectedCategory]);
```

Comment:
Is this really better UX? I think it's just better to just leave the Visual selected by default, even if it's empty.

Just a thought

Reply:
If you have no visual assets its worse UX landing on an empty page than landing on one with a correct filter selected